bgpd/bgp_filter.c

```diff
@@ -205,8 +205,18 @@ static struct as_list *as_list_new(void)
 
 static void as_list_free(struct as_list *aslist)
 {
-	XFREE(MTYPE_AS_STR, aslist->name);
-	XFREE(MTYPE_AS_LIST, aslist);
+	struct bp_as_excl_list *cur_bp = aslist->bp_list;
+	struct bp_as_excl_list *next_bp = NULL;
+
+	XFREE (MTYPE_AS_STR, aslist->name);
+	XFREE (MTYPE_AS_LIST, aslist);
```

Comment:
Shouldn't we free this after freeing aslist->bp_list?

Reply:
no need in this case, but i taken into account